digestAuth.go

```diff
@@ -17,28 +17,41 @@ type DigestAuth struct {
 	digestParts map[string]string
 }
 
-// Type identifies the DigestAuthenticator
-func (d *DigestAuth) Type() string {
-	return "DigestAuth"
+// Authorize the current request
+func (d *DigestAuth) Authorize(c *http.Client, rq *http.Request, method string, path string) error {
+	d.digestParts["uri"] = path
+	d.digestParts["method"] = method
+	d.digestParts["username"] = d.user
+	d.digestParts["password"] = d.pw
+	rq.Header.Set("Authorization", getDigestAuthorization(d.digestParts))
+	return nil
 }
 
-// User holds the DigestAuth username
-func (d *DigestAuth) User() string {
-	return d.user
+// Verify checks for authentication issues and may trigger a re-authentication
+func (d *DigestAuth) Verify(rq *http.Request, rs *http.Response, method string, path string) (reauth bool, err error) {
+	if rs.StatusCode == 401 {
```

Comment:
`http.StatusUnauthorized` instead of `401` maybe?

Reply:
on the other hand everywhere else numeric codes are used. so maybe better keep the numeric code.

Reply:
It's not 100% consistent. But the numbers dominate, not just iIt's not 100% consistent. But the numbers are predominant, not just in this code base.n this code base. I would stick with them.